src/polynomial/univariate.rs

```diff
@@ -0,0 +1,98 @@
+use std::iter;
+
+use ff::Field;
+
+/// Represents a univariate polynomial
+///
+/// Coefficients of the polynomial are presented from smaller to larger
+pub struct UnivariatePoly<F: Field>(pub(crate) Box<[F]>);
```

Comment:
what efficiency can be gained by using box?

Reply:
There are two arguments:

1. This type does not differ from `Vec` except that capacity is always equal to length. I.e. it is a must-have for large immutable collections, because there the additional space allocated by adding elements can be significant. But here, as far as I understand, we are not dealing with a large collection

2. We show that this collection is immutable. That is, it has no API for adding or removing items. So if we don't need to change coefficients, storing slice in heap is the default solution

There was even a video with a small explanation, if you are interested in more details https://www.youtube.com/watch?v=A4cKi7PTJSs.

Reply:
it's good to know. One more question based on the video: we might need clone for K,G,K. If so, Arc<[F]> is better?

Reply:
I've decided to add this as needed if we can't resolve it at the lifetime level

Reply:
I.e. migration to `Box` -> `Arc`  is more trivial than from `Vec` -> `Box`, so we can postpone it until we realize whether we need clones or not

Reply:
sounds good. You can add a TODO (no need to create an issue) in the comment if you might forget

Reply:
So the method does not implement `Clone` now, so it will be hard to forget if Clone is needed, no need for an extra reminder.

Also after our discussion I realized that the size of this polynomial can be large, so this adds arguments for using the Box\Arc approach